### PR TITLE
Add plugin list page

### PR DESCRIPTION
**Pull request by superstar54**
We foresee more plugins coming, and they may be developed by other developers and in other repositories. To help users easily find and manage the plugins, this PR adds a new plugin list page.

## Plugin registry
User can add their plugin entry point into the `plugins.yaml` file in this repository.

## Home page
On the home page, I added a new **Preference** section on the right side of the logo. For the moment, there is only the `plugin list` item, but we can add more user preference items in the future.

## Plugin list page
The UI has the following features:
- **Accordion Display for Plugins**: Each plugin is listed within an accordion-style component. This design keeps the interface clean and organized, allowing users to quickly scan through available plugins. The accordion's title bar displays essential information:
  - Plugin Name: Clearly indicates the name of the plugin for easy identification.
  - Installation Status: An icon indicator shows whether the plugin is currently installed.
- **Expandable Details**: Users can interact with the accordion to expand it, revealing more detailed information about each plugin.
- **Install button and remove button.**

Despite the `AppStore`'s extensive features, I opted against using it due to its complexity and lack of fit for this specific application.

## Demo
https://github.com/aiidalab/aiidalab-qe/assets/11457659/e1c37567-e1e6-486a-8083-906da8f5e851

## Future improvement
After this PR, open issues on
- [ ] version control, add `Update` button.
- [ ] add search bar

### CODE REVIEW

**Review by danielhollas — COMMENTED** [review]
Sorry, I will not have time to review this fully.

My only quick thought is that this seem overengineered and will be yet another thing to maintain. Do we really need another registry just for the QeApp? How about simply having a static yaml file in this repo?

**Comment by superstar54** [comment]
>  Do we really need another registry just for the QeApp? How about simply having a static yaml file in this repo?

Good point. For the QE App, we don't really need a separate repo like AiiDAlab and AiiDA.

**Comment by superstar54** [comment]
@danielhollas @unkcpz, any further comment on this PR? If not, I will merge it to be included in the next release. It is important that users can discover the plugins through this, e.g., @mikibonacci 's plugins.

Since this belongs to the `plugin management` work and will not affect the app's core, we can improve it in the future.

**Review by danielhollas — COMMENTED** [review]
I don't have time to do full review and as I am not an active user/developer of QeApp it's not my place. Hopefully @unkcpz has time to chime in.

Nevertheless, this looks cool and interesting. Thanks for moving the YAML config here.

**Comment by superstar54** [comment]
@mikibonacci feel free to modify the `plugins.yaml` files for your plugins.

**Comment by superstar54** [comment]
Hi all, if there are no more comments, I will merge this PR by the end of the day.

File: plugins.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+aiida-bader:
+    description: AiiDA plugin for the Bader analysis
+    author: Xing Wang
+    github: https://github.com/superstar54/aiida-bader
+    documentation: https://aiida-bader.readthedocs.io/
+    pip: aiida-bader
+
+aiidalab-qe-vibroscopy:
+    description: Plugin to compute vibrational properties of materials via the aiida-vibroscopy AiiDA plugin
+    author: Miki Bonacci
```
Comment (mikibonacci) [inline plugins.yaml:11]: can you please add also Andres Ortega Guerrero? then for me it is ok to merge

Comment (superstar54) [reply plugins.yaml]: Sure, Done!

Comment (mikibonacci) [reply plugins.yaml]: thanks a lot for the great work! it looks very nice

**Pull request merged**